src/HealthChecks.AzureStorage/AzureBlobStorageHealthCheck.cs

```diff
@@ -7,35 +7,34 @@ namespace HealthChecks.AzureStorage
 {
     public class AzureBlobStorageHealthCheck : IHealthCheck
     {
-        private readonly TokenCredential? _azureCredential;
-        private readonly Uri? _blobServiceUri;
-
-        private readonly string? _connectionString;
+        private readonly Func<BlobServiceClient> _clientFactory;

```

Comment:
I want to preserve the existing functionality, so I tried encapsulating it in a factory member. Alternatively, we could probably create two derived classes that fetch their respective clients differently.

Reply:
Factory method is fine.